### PR TITLE
Add XMLUnit as a managed dependency and to the test starter

### PR DESCRIPTION
**Pull request by andrew-j-cohen**
Signed-off-by: Andrew Cohen <acohen74@gmail.com>

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@andrew-j-cohen Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=9709)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=9709) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@andrew-j-cohen Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=9709)!

File: spring-boot-dependencies/pom.xml [hunk 1]
```diff
@@ -2472,6 +2473,11 @@
 				<artifactId>xml-apis</artifactId>
 				<version>${xml-apis.version}</version>
 			</dependency>
+			<dependency>
```
Comment (wilkinsona) [inline spring-boot-dependencies/pom.xml]: Could you move this up a bit please? We try to keep the entries in alphabetical order using `groupId` first and then `artifactId` if required

File: spring-boot-starters/spring-boot-starter-test/pom.xml [hunk 2]
```diff
@@ -40,6 +40,10 @@
 			<artifactId>junit</artifactId>
 		</dependency>
 		<dependency>
+			<groupId>org.xmlunit</groupId>
```
Comment (wilkinsona) [inline spring-boot-starters/spring-boot-starter-test/pom.xml]: Could you move this down a bit please? We try to keep the entries in alphabetical order using groupId first and then artifactId if required

**Comment by andrew-j-cohen** [comment]
@wilkinsona Thanks for the feedback. I've put the dependencies in the proper order.

**Comment by wilkinsona** [comment]
Thank you

**Comment by wilkinsona** [comment]
I missed something earlier. We should use `xmlunit2.version` for the property as we have others projects (Spring Web Services for example) that still depend on `xmlunit:xmlunit`. As a result, the Platform has version properties for both.

**Pull request closed without merging**